pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2041,7 +2045,11 @@ func buildCompleteFilterChain(mutable *istionetworking.MutableObjects, opts buil
 		chain := mutable.FilterChains[i]
 		opt := opts.filterChainOpts[i]
 		mutable.Listener.FilterChains[i].Metadata = opt.metadata
-		mutable.Listener.FilterChains[i].Name = opt.filterChainName
+		if len(opt.filterChainName) == 0 {
+			mutable.Listener.FilterChains[i].Name = FilterChainNamePrefix + uuid.New().String()
```

Comment:
Would it be useful if we add listener name to the filter chain name? and if we add it just append incrementing counter for each fc instead of generating uuid?

Reply:
> Would it be useful if we add listener name to the filter chain name? and if we add it just append incrementing counter for each fc instead of generating uuid?

Thanks a for the comments!
My two cents:
1) I think it would be somehow useful to add listener name to the filter chain name.
2) As for the id generation, IMHO, I would prefer uuid to increment counter if the key concern here is not performance. Using increment counter might bring more concerns to e.g. stateful chain name, security (info/status disclosure) and  restarted pilot etc.

Reply:
Anyway, let's see if there's further input from the community and I'm open to it.

Reply:
I think ideally we are adding a filter chain name to the real generate filterchains. So its useful in real world, not just our tests. Note it must be deterministic then

Reply:
The filter chain names can not be same, so i think this is why not set at beginning.